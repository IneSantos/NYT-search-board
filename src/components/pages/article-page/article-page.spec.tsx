import React from "react";
import { render } from '@testing-library/react';
import ArticlePage from "./article-page";

const mockedArticle = {
    id: "",
    headline: "",
    abstract: "", 
    keywords: [],
    section: "",
    snippet: "",
    pub_date: "",
    source: "",
    web_url: ""
};


describe("ArticlePage atom tests", () => {
  test(`ArticlePage component render`, () => {
    const { container } = render(
      <ArticlePage article={mockedArticle}/>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
