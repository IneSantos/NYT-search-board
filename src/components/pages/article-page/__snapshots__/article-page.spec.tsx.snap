// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePage atom tests ArticlePage component render 1`] = `
<section>
  <header>
    <div
      class="banner"
    >
      “The New York Times” article search application
    </div>
  </header>
  <div
    class="page-body-wrapper"
  >
    <div
      class="page-body"
    >
      <div
        class="article-header-wrapper"
      >
        <a
          class="article-action"
          href="/"
        >
           
          &lt; Go to results page
           
        </a>
        <div
          class="article-header"
        >
          <h3
            class="title"
          />
          <div
            class="subtitle"
          >
            NaN.NaN.NaN
          </div>
        </div>
      </div>
      <div
        class="article-body"
      >
        <div
          class="description"
        />
        <a
          class="article-action"
          href=""
        >
           
          Read the full article
           
        </a>
      </div>
    </div>
    <div
      class="page-footer"
    >
      <div
        class="css-1qq1qhc"
      >
        
        <div
          class=" rcs-buttons"
        >
          <button
            aria-label="Share on Twitter"
            class="css-trk4o9 rcs-button"
          >
            <i
              class="fab fa-twitter-square"
            />
          </button>
          <button
            aria-label="Share on Facebook"
            class="css-hn5fd rcs-button"
          >
            <i
              class="fab fa-facebook-square"
            />
          </button>
          <button
            aria-label="Share on Email"
            class="css-1pclnts rcs-button"
          >
            <i
              class="fas fa-envelope-square"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</section>
`;
